File: backend/app/schemas/vehicle.py
from typing import Optional

from pydantic import BaseModel

from app.models.enums import VehicleStatus


class VehicleCreate(BaseModel):
    reg_no: str
    name: str
    type: str
    capacity_kg: float
    odometer: float = 0
    acquisition_cost: float


class VehicleUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[str] = None
    capacity_kg: Optional[float] = None
    odometer: Optional[float] = None
    acquisition_cost: Optional[float] = None
    status: Optional[VehicleStatus] = None


class VehicleResponse(BaseModel):
    id: int
    reg_no: str
    name: str
    type: str
    capacity_kg: float
    odometer: float
    acquisition_cost: float
    status: VehicleStatus

    class Config:
        from_attributes = True

File: frontend/src/types/vehicle.ts
export type VehicleStatus = 'Available' | 'On Trip' | 'In Shop' | 'Retired'

export interface Vehicle {
  id: number
  reg_no: string
  name: string
  type: string
  capacity_kg: number
  odometer: number
  acquisition_cost: number
  status: VehicleStatus
}

export interface VehicleCreate {
  reg_no: string
  name: string
  type: string
  capacity_kg: number
  odometer: number
  acquisition_cost: number
}